File: componet/src/main/java/org/zgl/service/server/hall/GeneralizeOperation.java
package org.zgl.service.server.hall;

import org.zgl.dto.clinet.commond.GoldBaseDto;
import org.zgl.dto.clinet.hall.GeneralizeAwardDto;
import org.zgl.dto.clinet.hall.GeneralizeListDto;
import org.zgl.proxy.MethodDesc;
import org.zgl.proxy.rule.ITcpAsyncService;

/**
 * @author： 猪哥亮
 * @创建时间： 2018/8/21
 * @文件描述：
 */
public interface GeneralizeOperation extends ITcpAsyncService {
    @MethodDesc("打开推广列表")
    GeneralizeListDto openGeneralizeList();
    @MethodDesc("邀请")
    GeneralizeAwardDto invite(long targetUid);
    @MethodDesc("领取奖励")
    GoldBaseDto receiveAward();
}

File: cs/proxy/hall/GeneralizeOperation.cs
/// <summary>
/// 
/// </summary>
namespace org.zgl.service.server.hall{
	[Proxy(gameId = -10)]
	public interface GeneralizeOperation : ITcpAsyncService { 
		/// <summary>
		/// 邀请
		/// </summary>
		void invite(long targetUid);
		/// <summary>
		/// 邀请 ----> <<invite();回调>>
		/// </summary>
		void invite2CallBack(GeneralizeAwardDto callBackParam);
		/// <summary>
		/// 打开推广列表
		/// </summary>
		void openGeneralizeList();
		/// <summary>
		/// 打开推广列表 ----> <<openGeneralizeList();回调>>
		/// </summary>
		void openGeneralizeList2CallBack(GeneralizeListDto callBackParam);
		/// <summary>
		/// 领取奖励
		/// </summary>
		void receiveAward();
		/// <summary>
		/// 领取奖励 ----> <<receiveAward();回调>>
		/// </summary>
		void receiveAward2CallBack(GoldBaseDto callBackParam);
	}
}
